troposphere/static/js/components/projects/detail/resources/tableData/instance/Name.react.js

```diff
@@ -15,7 +15,7 @@ export default React.createClass({
       var instance = this.props.instance,
           name = instance.get('name').trim() || "[no instance name]";
 
-      if (!instance.id) {
+      if (!instance && !instance.get('id')) {
```

Comment:
... on another note. With the propTypes, we have some assurance that `instance.isNew()` or `instance.id` would be sufficient.